proposals/apple-mdm/Fleet-MDM-Plan.md

```diff
@@ -0,0 +1,218 @@
+
+The PoC can be found in: https://github.com/fleetdm/fleet/tree/apple-mdm.
+
+## High-level architecture
+
+### MDM functionality and services
+
+We will make use of the most widely-used open-source MDM project: [MicroMDM](https://github.com/micromdm).
+
+To implement an MDM server, we will embed the following three "nanoMDM" modules into Fleet:
+1. SCEP service: https://github.com/micromdm/scep
+2. MDM core: https://github.com/micromdm/nanomdm
+3. DEP functionality: https://github.com/micromdm/nanodep
+
+Such modules contain all the functionality and APIs/endpoints needed to implement an MDM server.
+
+For more information around "micro" vs "nano" MDM, see [*Introducting NanoMDM*](https://micromdm.io/blog/introducing-nanomdm/).
+
+Architecture of the current PoC:
+```mermaid
+flowchart LR
+
+subgraph Fleet [Fleet Server]
+
+    direction TB;
+
+    enroll[".mobileconfig<br>/mdm/apple/api/enroll"];
+    subgraph nanoMDMModules ["nanoMDM modules"]
+        direction TB;
+        nanoSCEP["nanoSCEP<br>/mdm/apple/scep"];
+        nanoMDM["nanoMDM<br>/mdm/apple/mdm"];
+        nanoDEP["nanoDEP<br>/mdm/apple/dep/proxy"];
+    end
+    MunkiRepo["Munki Repository<br>/mdm/apple/munki/repo"];
+    subgraph MunkiPkg [Munki Package Server];
+        manifest["manifest<br>/mdm/apple/munki/manifest"]
+        munkitools["munkitools-*.pkg<br>(signed)<br>/mdm/apple/munki/pkg"]
+    end
+    subgraph MySQL
+        direction LR;
+        mdmAppleDB[(nanoMDM<br>Schemas)];
+        fleetDB[(fleet)];
+    end
+end
+
+ApplePush[https://api.push.apple.com];
+AppleDEP[https://mdmenrollment.apple.com];
+nanoDEP -- Apple MDM DEP API ----> AppleDEP;
+nanoMDM --> ApplePush;
+
+nanoDEP --> mdmAppleDB;
+nanoMDM --> mdmAppleDB;
+```
+
+## Plan for Dogfood-MVP
+
+### Setup
+
+See [Fleet MDM setup](./Fleet-MDM-Setup.md).
+
+### Authority
+
+Only global admins will be allowed to use MDM functionality.
+
+### MDM features
+
+#### Enrollments
+
+"Enrollments" hold some settings for devices that will be enrolled to MDM.
+The MDM "enrollments" will allow Fleet to automatically enroll devices to specific teams, which then allows for applying specific MDM settings (depending on the team).
+
+For Dogfood-MVP, Fleet will allow creating global enrollments only (team support will be added at a subsequent iteration).
+Users will be able to create the two following types of enrollments:
+- Global manual enrollment
+- Global DEP enrollment
+
+We'll have a new `apple_manual_enrollments` table with the following fields:
+- ID (to deduce an "Enroll URL")
+- Name
+- Configuration (JSON?)
+- Team (or global)
+- Authentication configuration (initially NULL, meaning not authenticated)
+
+We'll have a new `apple_dep_enrollments` table with the following fields:
+- ID
+- Name
+- Configuration (JSON?)
+- Team (or global)
+- Authentication configuration (initially NULL, meaning not authenticated)
+- Reference to `dep_names` table (will be NULL at first)
```

Comment:
What's dep_names? new table? It's not referenced/described above.

Reply:
This is an implementation detail. It's a table defined by one of the components we use, [nanodep](https://github.com/micromdm/nanodep).

I'll clarify this.